core/domain/user_query_domain_test.py

```diff
@@ -0,0 +1,164 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the domain objects relating to platform parameters."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+
+from core.domain import user_query_domain
+from core.tests import test_utils
+import feconf
+import utils
+
+
+class UserQueryTests(test_utils.GenericTestBase):
+    """Test for the PlatformParameter."""
+
+    def setUp(self):
+        super(UserQueryTests, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.user_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.user_query_params = user_query_domain.UserQueryParams(
+            inactive_in_last_n_days=20,
+            has_not_logged_in_for_n_days=None,
+            created_at_least_n_exps=None,
+            created_fewer_than_n_exps=None,
+            edited_at_least_n_exps=None,
+            edited_fewer_than_n_exps=None
+        ),
+        self.user_query = user_query_domain.UserQuery(
+            query_id='user_query_id',
+            query_params=self.user_query_params,
+            submitter_id=self.user_id,
+            query_status=feconf.USER_QUERY_STATUS_PROCESSING,
+            user_ids=[],
+            sent_email_model_id=None,
+            created_on=datetime.datetime.utcnow(),
+        )
+
+    def test_validate_passes_without_exception(self):
+        self.user_query.validate()
```

Comment:
I was thinking of having this pattern:

pre_step: assert the object is valid.
action: Do some changes.
final_check: Assert the model is invalid.

(This way we will ensure that the validation error is only raised because of the action step as it was valid before the action step.) 

Currently, in each test, we are assuming that the object is valid by default, because we have one test which validates the object without performing any action. So all the test now depends on that assumption. (Developers can remove the test which validates the object without performing any action in the future.) If we will have separate pre_action_validation in each test then each test will be independent.)

__Marking this as an optional comment.__

Reply:
Right, that makes sense, I have moved the validation to the `setUp` function, that should address your concern, right?